MidasCivil_Adapter/CRUD/Read/Loads/BarDifferentialTemperatureLoad.cs

```diff
@@ -0,0 +1,97 @@
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.IO;
+using BH.oM.Adapters.MidasCivil;
+using BH.Engine.Adapter;
+using BH.oM.Structure.Loads;
+using BH.oM.Structure.Elements;
+
+namespace BH.Adapter.MidasCivil
+{
+    public partial class MidasCivilAdapter
+    {
+        /***************************************************/
+        /**** Private Methods                           ****/
+        /***************************************************/
+
+        private List<ILoad> ReadBarDifferentialTemperatureLoads(List<string> ids = null)
+        {
+            List<ILoad> bhomBarDifferentialTemperatureLoads = new List<ILoad>();
+
+            List<Loadcase> bhomLoadcases = ReadLoadcases();
+            Dictionary<string, Loadcase> loadcaseDictionary = bhomLoadcases.ToDictionary(
+                        x => x.Name);
+            string[] loadcaseFolders = Directory.GetDirectories(m_directory + "\\TextFiles");
+            int i = 1;
+            foreach (string loadcaseFolder in loadcaseFolders)
+            {
+                string loadcase = Path.GetFileName(loadcaseFolder);
+                List<string> barDifferentialTemperatureLoadText = GetSectionText(loadcase + "\\BSTEMPER");
+                if (barDifferentialTemperatureLoadText.Count != 0)
+                {
+                    List<List<string>> barDifferntialTemperatureSets = new List<List<string>>();
+                    List<string> barID = new List<string>();
+                    List<int> loadIndexes = new List<int>();
+                    List<string> delimitted = new List<string>();
+                    for (int j = 0; j < barDifferentialTemperatureLoadText.Count; j++)
+                    {
+                        if (!barDifferentialTemperatureLoadText[j].Contains("ELEMENT"))
+                        {
+                            loadIndexes.Add(j);
+                            delimitted = barDifferentialTemperatureLoadText[j].Split(',').ToList();
+                            barID.Add(delimitted[0].Trim());
+                        }
+                    }
+                    //This is needed to iterate over all BarDifferentialTemperatureLoads as it is input over several lines
```

Comment:
If this comment is referring to the code block above, place the comment above it